app/services/search/location_builder.rb

```diff
@@ -38,7 +35,13 @@ def location_category_search?
 
   def initialize_location_polygon
     @location_polygon = LocationPolygon.with_name(@location_category)
-    @location_polygon_boundary = [@location_polygon.boundary] if @location_polygon.present?
+    if @location_polygon.present?
+      @location_polygon_boundary = if @buffer_radius.present?
+                                     [@location_polygon.buffers[@buffer_radius]]
+                                   else
+                                     [@location_polygon.boundary]
```

Comment:
Is this array wrapping for Algolia? (Is it what the comment in ImportPolygon is referring to?)

We might want to find a way to do the array wrapping only in one place, e.g. by writing it in the getter method (replacing attr_reader : location_polygon_boundary). But we haven't yet hit the 'Rule of Three': three strikes and refactor.

Reply:
I think it does only happens in the location builder.

Reply:
I meant lines 40 and 42. But this is an acceptable level of duplication, for me.